acceptance-tests/src/test/java/it/com/atlassian/bitbucket/jenkins/internal/SmokeTest.java

```diff
@@ -92,25 +133,114 @@ public void setUp() throws IOException {
         jenkinsConfig.configure();
         serverId = "bbs-" + randomUUID();
         new BitbucketPluginConfigArea(jenkinsConfig)
-                .addBitbucketServerConfig(serverId, BITBUCKET_BASE_URL, bbsPersonalToken.getId());
+                .addBitbucketServerConfig(serverId, BITBUCKET_BASE_URL, bbsPersonalAccessToken.getId());
         jenkinsConfig.save();
 
         // Fork repo
         forkRepo = forkRepository(PROJECT_KEY, REPO_SLUG, "fork-" + randomUUID());
+
+        // Log into Bitbucket
+        BITBUCKET.visit(BitbucketLoginPage.class).login(BITBUCKET_ADMIN_USERNAME, BITBUCKET_ADMIN_PASSWORD, DashboardPage.class);
     }
 
     @After
     public void tearDown() {
         // There's a limit of 100 personal access tokens per user in Bitbucket Server, hence we clean up the access
         // tokens after each test to ensure repeated test runs against the same Bitbucket instance does not fail.
-        if (bbsPersonalToken != null) {
-            deletePersonalToken(bbsPersonalToken.getId());
+        if (bbsPersonalAccessToken != null) {
+            deletePersonalAccessToken(bbsPersonalAccessToken.getId());
         }
         if (bbsSshCreds != null) {
             deleteSshPublicKey(bbsSshCreds.getId());
         }
+        if (applinkUrl != null) {
+            deleteApplink(applinkUrl);
+            applinkUrl = null;
+        }
+        if (forkRepo != null) {
+            deleteRepoFork(forkRepo.getSlug());
+            forkRepo = null;
+        }
     }
 
+    @Test
+    public void testRunBuildActionWtihFreestlyeJob() throws Exception {
```

Comment:
This test is huge, as we discussed I think it would benefit a lot from moving things out to methods that are not immediately test related. Things that come to mind are setting up applink, configuring the environment, authorise the server. You can then call those methods from here and the test is slimmer and easier to read.